src/SchemaRegistry.ts

```diff
@@ -57,23 +58,41 @@ export default class SchemaRegistry {
     this.options = options
   }
 
+  private isConfluentSchema(
+    schema: RawAvroSchema | AvroSchema | ConfluentSchema,
+  ): schema is ConfluentSchema {
+    return (schema as ConfluentSchema).schema != null
+  }
+
   private getConfluentSchema(
     schema: RawAvroSchema | AvroSchema | ConfluentSchema,
   ): ConfluentSchema {
     let confluentSchema: ConfluentSchema
     // convert data from old api (for backwards compatibility)
-    if (!(schema as ConfluentSchema).schemaString) {
+    if (!this.isConfluentSchema(schema)) {
       // schema is instanceof RawAvroSchema or AvroSchema
       confluentSchema = {
         type: SchemaType.AVRO,
-        schemaString: JSON.stringify(schema),
+        schema: JSON.stringify(schema),
       }
     } else {
       confluentSchema = schema as ConfluentSchema
     }
     return confluentSchema
   }
 
+  public async register(
+    schema: Exclude<ConfluentSchema, AvroConfluentSchema>,
+    userOpts: Opts,
+  ): Promise<RegisteredSchema>
+  public async register(
+    schema: RawAvroSchema | AvroConfluentSchema,
+    userOpts?: Omit<Opts, 'subject'> & { subject?: string },
```

Comment:
I'm not sure what's going on here with these multiple signatures.
I assume that you wanted to indicate that only Avro schemas with the v1 api don't have to specify a subject.
But then why does the 3rd signature below also have `RawAvroSchema`?
It would make more sense to me if we had 1 signature to define the old v1 api and 1 signature for the new v2 api. But maybe I just don't understand what you tried to do here.

Reply:
nevermind, I have just now read your comments above. you can ignore this comment :)